float32 posx
float32 posy
---
bool success
---
float32 progress